tests/api_test.py

```diff
@@ -69,3 +69,21 @@ def test_ignore_absolute_paths():
     assert get_path_if_relative(
         '/usr/lib/shared/schema.json',
     ) is None
+
+
+def test_resolve_nested_refs():
+    """
+    Make sure we resolve nested refs gracefully and not get lost in
+    the recursion. Also make sure we don't rely on dictionary order
+    """
+    import os
+    import yaml
+    from bravado_core.spec import Spec
+    from random import randint
+    from pyramid_swagger.api import resolve_refs
+    seed = randint(0, 4294967295)
```

Comment:
Ideally we'd find a value for which this particular spec fails on the old code, then hardcode that :)